protocol/requests/create_community_token_permission_request.go

```diff
@@ -0,0 +1,56 @@
+package requests
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+var (
+	ErrCreateCommunityTokenPermissionInvalidCommunityID    = errors.New("create community token permission needs a valid community id")
+	ErrCreateCommunityTokenPermissionTooManyTokenCriteria  = errors.New("too many token criertia")
+	ErrCreateCommunityTokenPermissionInvalidPermissionType = errors.New("invalid community token permission type")
+	ErrCreateCommunityTokenPermissionInvalidTokenCriteria  = errors.New("invalid community permission token criteria data")
+)
+
+type CreateCommunityTokenPermission struct {
+	CommunityID   types.HexBytes                         `json:"communityId"`
+	Type          protobuf.CommunityTokenPermission_Type `json:"type"`
+	TokenCriteria []*protobuf.TokenCriteria              `json:"tokenCriteria"`
+}
+
+func (p *CreateCommunityTokenPermission) Validate() error {
+	if len(p.CommunityID) == 0 {
+		return ErrCreateCommunityTokenPermissionInvalidCommunityID
+	}
+
+	if len(p.TokenCriteria) > 5 {
```

Comment:
Should we move this to a constant somewhere so we can modify it only in one place in case we need this type of check somewhere else?

Reply:
Yes, I'll make it a constant